app/control.py

```diff
@@ -1,20 +1,81 @@
 #!/usr/bin/env python3
 
-import smartthings
-import qcells
 import json
+import datetime
+import time
+import deviceFactory
+import inverterFactory
+import sys
+import traceback
 
-with open("config.json") as configuration:
+if(len(sys.argv) > 1):
+    configFile = arg1 = sys.argv[1]
+else:
+    configFile = "config.json"
+
+with open(configFile) as configuration:
     jsonData = json.load(configuration)
+    factory = deviceFactory.deviceFactory()
     token = jsonData["smartthings"]["token"]
+    handsoffTime = datetime.timedelta(minutes=int(jsonData["handsOffTime"]))
     devices = {}
+    prios = [-1] # -1 as the "invalid" prio
     for device in jsonData["smartthings"]["devices"]:
-        devices[device["name"]] = smartthings.SmartthingDevice(device["id"],device["name"], token)
+        prio = device["prio"]
+        if(prio not in prios):
+            prios.append(prio)
+        devices[device["name"]] = factory.addDevice(token, device)
 
+    prios.sort()
+    priolevels = len(prios)
+    currentPrioIndex = 0
     qcellJson = jsonData["qcells"]["devices"][0]
-    wechselrichter = qcells.qcellDevice(qcellJson["token"], qcellJson["sn"])
+    inverterFactory = inverterFactory.inverterFactory()
+    wechselrichter = inverterFactory.addDevice(qcellJson)
+    print("Initialization successfull")
```

Comment:
Refactor Idee: Initialisierung in eine eigene "config-Klasse" und auf die Instanz davon zugreifen. Macht auch den Check auf eine korrekte Konfiguration einfacher

Reply:
resolved